uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/absence-tab/components/contract-entitlements_test.js

```diff
@@ -0,0 +1,109 @@
+/* eslint-env amd, jasmine */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'common/angular',
+  'common/angularMocks',
+  'leave-absences/shared/config',
+  'leave-absences/absence-tab/app',
+  'common/mocks/services/api/contract-mock',
+  'mocks/apis/absence-type-api-mock',
+  'leave-absences/shared/modules/shared-settings'
+], function (_, moment, angular) {
+  'use strict';
+
+  describe('contactEntitlements', function () {
+    var $compile, $log, $rootScope, $q, component, controller, Contract,
+      ContractAPIMock, AbsenceType, AbsenceTypeAPIMock;
+
+    beforeEach(module('leave-absences.templates', 'absence-tab',
+                      'common.mocks', 'leave-absences.mocks'
+    ));
+
+    beforeEach(inject(['api.contract.mock', 'AbsenceTypeAPIMock', function (_ContractAPIMock_, _AbsenceTypeAPIMock_) {
+      ContractAPIMock = _ContractAPIMock_;
+      AbsenceTypeAPIMock = _AbsenceTypeAPIMock_;
+    }]));
+
+    beforeEach(inject(function (_$compile_, _$log_, _$rootScope_, _$q_, _Contract_, _AbsenceType_) {
+      $compile = _$compile_;
+      $log = _$log_;
+      $rootScope = _$rootScope_;
+      $q = _$q_;
+
+      Contract = _Contract_;
+      AbsenceType = _AbsenceType_;
+
+      spyOn($log, 'debug');
+      spyOn(AbsenceType, 'all').and.callFake(function () {
```

Comment:
The think the below is sufficient? Because `AbsenceTypeAPIMock.all` returns a promise too.
`spyOn(AbsenceType, 'all').and.callFake(AbsenceTypeAPIMock.all);`